From cbf59f872c608c4eafe89407aa0d0fe908ebbe09 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:18 -0400
Subject: [usb] core: fix read of usbdevfs_ctrltransfer fields in proc_control()

Message-id: <1410887463-20674-7-git-send-email-dzickus@redhat.com>
Patchwork-id: 95346
O-Subject: [RHEL7 PATCH 006/591] usbcore: fix read of usbdevfs_ctrltransfer fields in proc_control()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit c8f2efc8f636506e0f0c2ba4035382076875f0c1
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Sat Aug 31 18:09:14 2013 +0300

    usbcore: fix read of usbdevfs_ctrltransfer fields in proc_control()

    Urb fields are stored in struct usbdevfs_ctrltransfer in CPU byteorder
    and not in little endian, so there is no need to be converted.
    This bug was reported by sparse.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/devio.c b/drivers/usb/core/devio.c
index 71dc5d7..bd429ea 100644
--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -914,10 +914,8 @@ static int proc_control(struct dev_state *ps, void __user *arg)
  snoop(&dev->dev, "control urb: bRequestType=%02x "
   "bRequest=%02x wValue=%04x "
   "wIndex=%04x wLength=%04x\n",
-  ctrl.bRequestType, ctrl.bRequest,
-  __le16_to_cpup(&ctrl.wValue),
-  __le16_to_cpup(&ctrl.wIndex),
-  __le16_to_cpup(&ctrl.wLength));
+  ctrl.bRequestType, ctrl.bRequest, ctrl.wValue,
+  ctrl.wIndex, ctrl.wLength);
  if (ctrl.bRequestType & 0x80) {
   if (ctrl.wLength && !access_ok(VERIFY_WRITE, ctrl.data,
             ctrl.wLength)) {
-- 
1.7.1